### PR TITLE
Big clean-up of driver return codes (#1794)

### PR DESCRIPTION
**Pull request by polluks2**
Now nonobvious code should contain an assert.

### CODE REVIEW

File: libsrc/apple2/mou/a2.stdmou.s [hunk 1]
```diff
@@ -133,8 +133,8 @@ next:   inc     ptr1+1
         bcc     :+
 
         ; Mouse firmware not found
-        lda     #<MOUSE_ERR_NO_DEVICE
-        ldx     #>MOUSE_ERR_NO_DEVICE
+        lda     #MOUSE_ERR_NO_DEVICE
+        ldx     #>$0000
```
Comment (groessler) [inline libsrc/apple2/mou/a2.stdmou.s:137]: Why did you change this (and the other similar ones)? Looks OK to me. Your change doesn't look nicer (worse in my eyes), and it breaks if MOUSE_ERR_NO_DEVICE is >= 0x100.

Comment (polluks2) [reply libsrc/apple2/mou/a2.stdmou.s:137]: Please look at our long discussion, MOUSE_ERR_NO_DEVICE >= 0x100 is impossible because the return value is char.

Comment (groessler) [reply libsrc/apple2/mou/a2.stdmou.s:137]: Ok. (I don't remember that specific discussion right now.)
But "#>$0000" for just "0" looks really ugly to me.

Comment (polluks2) [reply libsrc/apple2/mou/a2.stdmou.s:137]: This is an intended stumbling block because it isn't a high byte of an int.

**Comment by mrdudz** [comment]
> > But "#>$0000" for just "#0" looks really ugly to me.

> This is an intended stumbling block because it isn't a high byte of an int but integer promotion

I have to admit i find this particular thing ugly too. The reason is demonstrated right here: it is not obvious what this is about. Actually not only that, it even looks like something is wrong there. I'd prefer something like this:
```
lda #FOO
ldx #0 ; return value is char
```
... because it achives the same goal and is a lot easier to read/understand.

**Comment by polluks2** [comment]
Well, this patch is just a unification. The "ugly" syntax is already common, please try
`grep -r "#>\$0000" libsrc/`

**Comment by mrdudz** [comment]
> The "ugly" syntax is already common

That doesn't mean it can't be changed though (and it should be easy too with some inline sed)

**Comment by polluks2** [comment]
Feel free to use sed after a merge (don't forget adjusting the comments).

**Comment by mrdudz** [comment]
> Feel free to use sed after a merge (don't forget adjusting the comments).

I am not willing to merge a PR that requires me to fix it after that, sorry. I am already working in other corners of the project and my time is limited. This is exactly why you should discuss and ask before doing larger changes (and its exactly why i started contributing.md) - it makes everyones life easier.

**Comment by groessler** [comment]
I'm with @mrdudz.  @polluks2, please fix that yourself. But I very much like the non-"#>0000" changes.

**Comment by mrdudz** [comment]
closed via [ffa83c3](https://github.com/cc65/cc65/commit/ffa83c32a4d4cb2f7bd7749aa3e2766c370e60a7)
